### ISSUE TITLE
I want to control the light position and intensity during training.How can I optimize the light parameters like tf-mesh-render, which has a 27-dimensioanl parameter and can be optimized during training ?

### ISSUE THREAD

**Issue by tuanhui-li** [facebookresearch/pytorch3d#338]

**Comment by gkioxari** [comment]
Light parameters are torch.tensors so you can optimize them as any other tensor.

**Comment by tuanhui-li** [comment]
@gkioxari so can you give a simple demo code for how to optimize the light parameters ? I use the official demo code and change the light parameter to tensor, but the tensor can not be optimized.
```
location = torch.zeros(1,3, dtype=torch.float32)
location.requires_grad = True
lights = PointLights(device=device, location=location )

renderer = MeshRenderer(
    rasterizer=MeshRasterizer(
        cameras=cameras,
        raster_settings=raster_settings
    ),
    shader=TexturedSoftPhongShader(
        device=device,
        cameras=cameras,
        lights=lights
    )
)
```

**Comment by gkioxari** [comment]
Did you add `location` to the optimization parameters? 

```
optimizer = torch.optim.SGD([whatever_other_params, locatoin], lr=1.0, momentum=0.9)
```

**Comment by tuanhui-li** [comment]
> According to your demo code `location.grad` should be not None but for the tensor to be optimized you need to add `location` to the optimization parameters, for example
> 
> ```
> optimizer = torch.optim.SGD([whatever_other_params, location], lr=1.0, momentum=0.9)
'''
of course I add them to the optimizer, when I do backward, the grad of the location is none, maybe you can share a demo code like the code above if possible, looking forward to your reply !

**Comment by sbyNiceBlowJob** [comment]
I 'm encountering the same problem. I 'm optimizing the ambient color of the Directional light but the gradient turns out to be 0. Can you please tell me how you solve this problem? Thank you!
